src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,191 @@
+     */
+    public static Command parse(String command) throws DukeException {
+        if (command.equals("bye")) {
+            return new ExitCommand();
+        } else if (command.equals("list")) {
+            return new ListCommand();
+        } else {
+            String[] task = command.split(" ");
+            String instruction = task[0];
+
+            if (instruction.equals("done")) {
+                int taskNumber = Integer.parseInt(task[1]);
+                return new DoneCommand(taskNumber - 1);
+            } else if (instruction.equals("delete")) {
+                try {
+                    int taskNumber = Integer.parseInt(task[1]);
+                    return new DeleteCommand(taskNumber - 1);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new DukeException("OOPS!!! Provide a task number to delete!");
+                }
+            } else {
+                switch (instruction) {
+                case "todo": {
+                    try {
+                        todoCheck(task);
+                        Task todo = new Todo(command.substring(5));
+                        return new AddCommand(todo);
+                    } catch (DukeException e) {
+                        System.err.println("Something went wrong: " + e.getMessage());
+                        break;
+                    }
+                }
+                case "deadline": {
+                    try {
+                        deadlineCheck(task, command);
+                        Task deadline = new Deadline(command.substring(9, command.indexOf("/by") - 1),
+                                dateFormatter(command.substring(command.indexOf("/by") + 4)));
+                        return new AddCommand(deadline);
+                    } catch (DukeException e) {
+                        System.err.println("Something went wrong: " + e.getMessage());
+                        break;
+                    }
+                }
+                case "event": {
+                    try {
+                        eventCheck(task, command);
+                        Task event = new Event(command.substring(6, command.indexOf("/at") - 1),
+                                dateFormatter(command.substring(command.indexOf("/at") + 4)));
+                        return new AddCommand(event);
+                    } catch (DukeException e) {
+                        System.err.println("Something went wrong: " + e.getMessage());
+                        break;
+                    }
+                }
+                    case "find": {
+                        try {
+                            findCheck(task);
+                            return new FindTaskCommand(command.substring(5).trim());
+                        } catch (DukeException e) {
+                            System.err.println("Something went wrong: " + e.getMessage());
+                            break;
+                        }
+                    }
```

Comment:
Just noticed that you enclosed each of the case blocks in braces. Not necessary to do so

Reply:
Does it affect the performance of the programme? :O Thanks for this though!

Reply:
I don't think it affects the performance. Maybe it is my first time seeing braces being used to enclose an environment within each case statement, so I had a difficult time looking for the matching braces :x